gems/aws-sdk-core/spec/aws-sdk-core/config-file-sso-session-parser-tests.json

```diff
@@ -0,0 +1,720 @@
+      }
+    },
+
+    {
+      "name": "Continuations cannot be used outside of a property.",
+      "input": {
+        "configFile": "[sso-session foo]\n -continued"
+      },
+      "output": {
+        "errorContaining": "Expected a property definition"
+      }
+    },
+
+    {
+      "name": "Continuations reset with sso-session definitions.",
+      "input": {
+        "configFile": "[sso-session foo]\nname = value\n[sso-session foo]\n -continued"
+      },
+      "output": {
+        "errorContaining": "Expected a property definition"
+      }
+    },
+
+    {
+      "name": "Duplicate sso-sessions in the same file merge properties.",
+      "input": {
+        "configFile": "[sso-session foo]\nname = value\n[sso-session foo]\nname2 = value2"
+      },
+      "output": {
+        "ssoSessions": {
+          "foo": {
+            "name": "value",
+            "name2": "value2"
+          }
+        }
+      }
+    },
+
+    {
+      "name": "Duplicate properties in an sso-session use the last one defined.",
+      "input": {
+        "configFile": "[sso-session foo]\nname = value\nname = value2"
+      },
+      "output": {
+        "ssoSessions": {
+          "foo": {
+            "name": "value2"
+          }
+        }
+      }
+    },
+
+    {
+      "name": "Duplicate properties in duplicate sso-sessions use the last one defined.",
+      "input": {
+        "configFile": "[sso-session foo]\nname = value\n[sso-session foo]\nname = value2"
+      },
+      "output": {
+        "ssoSessions": {
+          "foo": {
+            "name": "value2"
+          }
+        }
+      }
+    },
+
+    {
+      "name": "Invalid sso-session names are ignored.",
+      "input": {
+        "configFile": "[sso-session in valid]\nname = value"
+      },
+      "output": {
+        "ssoSessions": {}
+      }
+    },
+
+    {
+      "name": "Invalid property names are ignored.",
+      "input": {
+        "configFile": "[sso-session foo]\nin valid = value"
+      },
+      "output": {
+        "ssoSessions": {
+          "foo": {}
+        }
+      }
+    },
+
+    {
+      "name": "All valid identifier characters are supported.",
+      "input": {
+        "configFile": "[sso-session ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789_-/.%@:+]"
+      },
+      "output": {
+        "ssoSessions": {
+          "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789_-/.%@:+": {}
+        }
+      }
+    },
+
+    {
+      "name": "All valid property name characters are supported.",
+      "input": {
+        "configFile": "[sso-session foo]\nABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789_-/.%@:+ = value"
+      },
+      "output": {
+        "ssoSessions": {
+          "foo": {
+            "abcdefghijklmnopqrstuvwxyzabcdefghijklmnopqrstuvwxyz0123456789_-/.%@:+": "value"
+          }
+        }
+      }
+    },
+
+    {
+      "name": "Properties can have sub-properties.",
```

Comment:
I'm not sure that sub properties in sso-sessions should be supported?

Reply:
I'm not sure either. But it says they can have properties. Properties can have sub properties. Transitively it means it is supported. Supporting sub properties in session is simpler code but it's not used currently.

Reply:
Yeah that makes sense.  I think its reasonable to leave since it keeps the code simpler.